### PR TITLE
new package: ecmwf-atlas

### PR DESCRIPTION
**Pull request by eugeneswalker**
CC @climbfuji @srherbener

### CODE REVIEW

**Review by climbfuji — CHANGES_REQUESTED** [review]
This package does not match what is in our spack-stack fork of spack, looking at the diffs it is quite behind. Can you update it from https://github.com/JCSDA/spack/tree/610a6c4fee272a9d06e83b3cd962542fe8280219/var/spack/repos/builtin/packages/ecmwf-atlas/ please?

**Comment by eugeneswalker** [comment]
> This package does not match what is in our spack-stack fork of spack, looking at the diffs it is quite behind. Can you update it from https://github.com/JCSDA/spack/tree/610a6c4fee272a9d06e83b3cd962542fe8280219/var/spack/repos/builtin/packages/ecmwf-atlas/ please?

Done.

**Review by climbfuji — COMMENTED** [review]
Perfect, thank you. This matches what we have in our fork. Looking at the CI failures we will need to update eckit, fckit and a few other packages related to it (ectrans, fiat) as well.

I can take care of those or you can, whatever works best for you.

**Comment by eugeneswalker** [comment]
> Perfect, thank you. This matches what we have in our fork. Looking at the CI failures we will need to update eckit, fckit and a few other packages related to it (ectrans, fiat) as well.
> 
> I can take care of those or you can, whatever works best for you.

Yes, I'll do the updates here and when CI is all passing, I'll only ping you then.

**Review by srherbener — APPROVED** [review]
I verified that the version updates of all the packages here cover what we are requesting in spack-stack@1.5.0. I think it all looks good now, but I haven't tested (relying on CI testing).

**Review by climbfuji — APPROVED** [review]
Thanks very much for this @eugeneswalker!

File: var/spack/repos/builtin/packages/eckit/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/eckit/package.py]: Why adding a variant to raise an InstallError during the build? Can't we just hard-code that `BUILD_SHARED_LIBS` is `ON`?

File: var/spack/repos/builtin/packages/ecmwf-atlas/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/ecmwf-atlas/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/fiat/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/fiat/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/fiat/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/fiat/package.py]:
```suggestion
```

**Review by alalazo — CHANGES_REQUESTED** [review]
A few first comments

File: var/spack/repos/builtin/packages/eckit/package.py
Comment (eugeneswalker) [reply var/spack/repos/builtin/packages/eckit/package.py]: What do you think @climbfuji?

Comment (climbfuji) [reply var/spack/repos/builtin/packages/eckit/package.py]: Changing this is fine with me, either logic reminds us (me) that we have to implement this at some point.

File: var/spack/repos/builtin/packages/ecmwf-atlas/package.py
Comment (climbfuji) [reply var/spack/repos/builtin/packages/ecmwf-atlas/package.py]: I am against removing this, because it takes time to go through CMakeLists.txt and check with options are available in principle. I don't want to do this again, therefore the reminder that cgal is an option that we simply haven't implemented yet.

File: var/spack/repos/builtin/packages/fiat/package.py
Comment (climbfuji) [reply var/spack/repos/builtin/packages/fiat/package.py]: Same as for cgal above.

File: var/spack/repos/builtin/packages/fiat/package.py
Comment (climbfuji) [reply var/spack/repos/builtin/packages/fiat/package.py]: Same as for cgal above

File: var/spack/repos/builtin/packages/eckit/package.py
Comment (eugeneswalker) [reply var/spack/repos/builtin/packages/eckit/package.py]: @alalazo Just pushed the requested changes.

**Review by climbfuji — COMMENTED** [review]
I dismissed my approval because I don't want the variants-to-be-implemented being removed, sorry.

**Comment by alalazo** [comment]
> I dismissed my approval because I don't want the variants-to-be-implemented being removed, sorry

What is the added value of commented code? You can still implement the variants later.

**Comment by climbfuji** [comment]
> > I dismissed my approval because I don't want the variants-to-be-implemented being removed, sorry
> 
> What is the added value of commented code? You can still implement the variants later.

Yes of course, but I need to go back and look at the CMakeLists.txt files to remember which variants. Of course I can create a github issue for that if that's preferred.

**Review by climbfuji — APPROVED** [review]
Although I'm no longer sure what the role of a package maintainer is,, I'm not going to hold up this PR. I created an issue to remind myself about the variants that need to be implemented for which the inline comments get stripped out as part of this PR (https://github.com/JCSDA/spack-stack/issues/783).

**Comment by alalazo** [comment]
> Although I'm no longer sure what the role of a package maintainer is, I'm not going to hold up this PR.

What should it be, in your opinion, and how suggesting to delete commented out code goes against that role? 

> I am against removing this, because it takes time to go through CMakeLists.txt and check with options are available in principle. I don't want to do this again, therefore the reminder that cgal is an option that we simply haven't implemented yet.

Not having to look at `CMakeLists.txt` ever again is very likely a false hope. If a new version is added, `CMakeLists.txt` might change too - so having a look at that might be needed. 

Also, usually, checking the config options means just running the configuration. By doing:
```
$ spack install --only dependencies ecmwf-atlas
$ spack stage ecmwf-atlas
$ spack cd ecmwf-atlas
$ spack build-env ecmwf-atlas --bash
$ mkdir build && cd build
$ cmake ..
```
I'll get some configuration generated, then using `ccmake`:
Screenshot from 2023-09-23 14-49-38

I can inspect which options are used without losing a lot of time. Am I missing something specific to this package?

**Comment by eugeneswalker** [comment]
Thank you @climbfuji and @alalazo for the help with this! Let us proceed with merge when you have time to finalize @alalazo !

File: var/spack/repos/builtin/packages/ectrans/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/ectrans/package.py]: Can we have descriptions which are not questions?

File: var/spack/repos/builtin/packages/fiat/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/fiat/package.py]: Same here

**Review by alalazo — COMMENTED** [review]
@eugeneswalker The recipe basically LGTM, can you quickly push changes to the description of variants, so we can proceed with merging? 

If you have some time, it would be good to add a brief comment on top of patches too, to explain what they do and why they are required. That generally helps reading the recipe without switching contexts.

File: var/spack/repos/builtin/packages/ectrans/package.py
Comment (eugeneswalker) [reply var/spack/repos/builtin/packages/ectrans/package.py]: Yes, done.

**Comment by eugeneswalker** [comment]
> @eugeneswalker The recipe basically LGTM, can you quickly push changes to the description of variants, so we can proceed with merging?

Done!

> If you have some time, it would be good to add a brief comment on top of patches too, to explain what they do and why they are required. That generally helps reading the recipe without switching contexts.

I don't know what these patches do. These package definitions are coming from a NOAA fork and I think the people who know what these patches do are more than one. Perhaps we could save the addition of explanatory comments for patches to follow up PRs?

**Review by alalazo — APPROVED** [review]

**Comment by alalazo** [comment]
> Perhaps we could save the addition of explanatory comments for patches to follow up PRs?

:+1: Feel free to ping me when that is submitted.

**Pull request merged**
